packages/@aws-cdk/aws-redshift/lib/cluster.ts

```diff
@@ -662,4 +663,54 @@ export class Cluster extends ClusterBase {
       throw new Error('Cannot add a parameter to an imported parameter group.');
     }
   }
+
+  /**
+   * Adds a role to the cluster
+   *
+   * @param role the role to add
+   */
+  public addIamRole(role: iam.IRole): void {
+    const clusterRoleList = this.cluster.iamRoles ?? [];
+
+    if (clusterRoleList.includes(role.roleArn)) {
+      throw new Error('Role is already attached to the cluster');
```

Comment:
```suggestion
      throw new Error(`Role '${role.roleArn}' is already attached to the cluster`);
```

Reply:
Good idea. I don't know why I didn't implement this before. I've just done it